lib/rouge/guessers/disambiguation.rb

```diff
@@ -131,6 +131,12 @@ def match?(filename)
         next TeX if matches?(/\A\s*(?:\\|%)/)
         next Apex
       end
+
+      disambiguate '*.pp' do
+        next Pascal if matches?(/(function|begin|end\.)/)
```

Comment:
@Alexey-T do these 3 keywords representable for most Free Pascal code? 🤔

Reply:
@tancnle yes, they do. only add `\b` before and after.

Reply:
I have just noted that `\b` after does not work with non-word characters such as `end.` and `end;`. I have split these into two rules in https://github.com/rouge-ruby/rouge/pull/1845/commits/e8c225befa3e96d47e547f72d9f26a3c0672dae1 for better readability. Let me know what you think.